event_database/cloud_functions/process-transfer.go

```diff
@@ -62,26 +72,40 @@ func transformHexAddressToNative(chain vaa.ChainID, address string) string {
 		if val, ok := tokenAddressExceptions[address]; ok {
 			return val
 		}
-
-		trimmed := address[(len(address) - 40):]
-		data, decodeErr := hex.DecodeString(trimmed)
-		if decodeErr != nil {
-			fmt.Printf("failed to decode unpadded string: %v\n", decodeErr)
-		}
-		encodedAddr, convertErr := bech32.ConvertAndEncode("terra", data)
-		if convertErr != nil {
-			fmt.Println("convert error from cosmos bech32. err", convertErr)
-		}
-		return encodedAddr
+		return humanAddressTerra(address)
 	case vaa.ChainIDAlgorand:
 		// TODO
 		return ""
+	case vaa.ChainIDTerra2:
+		if addressType == AddressTypeContract {
+			// handle terra2 native assets manually
+			if val, ok := tokenAddressExceptions[address]; ok {
+				return val
+			}
+			// TODO: other terra2 asset types
```

Comment:
For now we are just supporting uluna on terra2 (that's the only asset in the allowlist anyway). But in the future we should make a contract query to get the native address.